src/audio/asrc/asrc.c

```diff
@@ -268,94 +201,7 @@ static void src_copy_s16(struct processing_module *mod,
 	*n_written = out_frames;
 }
 
-#ifndef CONFIG_IPC_MAJOR_4
-static inline uint32_t asrc_get_source_rate(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->source_rate;
-}
 
-static inline uint32_t asrc_get_sink_rate(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->sink_rate;
-}
-
-static inline uint32_t asrc_get_operation_mode(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->operation_mode;
-}
-
-static inline bool asrc_get_asynchronous_mode(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->asynchronous_mode;
-}
-#else
-static inline uint32_t asrc_get_source_rate(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return ipc_asrc->base.audio_fmt.sampling_frequency;
-}
-
-static inline uint32_t asrc_get_sink_rate(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return ipc_asrc->out_freq;
-}
-
-static inline uint32_t asrc_get_operation_mode(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return ipc_asrc->asrc_mode & (1 << IPC4_MOD_ASRC_PUSH_MODE) ? ASRC_OM_PUSH : ASRC_OM_PULL;
-}
-
-static inline bool asrc_get_asynchronous_mode(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return false;
-}
-
-#endif
-
-static int asrc_init(struct processing_module *mod)
```

Comment:
asrc_init is standard interface for module interface, need keep in asrc.c, only move out specific code with like asrc_common_init to cover both ipc3/ipc4 cases.